From c154aa771a31bf5cd6d648a02cf19b98a2910e81 Mon Sep 17 00:00:00 2001
From: Lorenzo Colitti <lorenzo@google.com>
Date: Mon, 16 Nov 2015 11:37:57 +0900
Subject: [PATCH] Don't enable MDNS on non-multicast, non-broadcast
 interfaces.

This is unlikely to work, and in the case of cell interfaces, it
may be harmful (e.g., use metered bandwidth, or compete for
resources with carrier traffic such as VoLTE).

Bug: 25629907
Bug: 25669326
Change-Id: I2622182ce1a3bb1c605de8afce61cf5ae1f4cbfb
---
 mDNSPosix/mDNSPosix.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mDNSPosix/mDNSPosix.c b/mDNSPosix/mDNSPosix.c
index 0e10bd5..0485924 100755
--- a/mDNSPosix/mDNSPosix.c
+++ b/mDNSPosix/mDNSPosix.c
@@ -989,7 +989,7 @@ mDNSlocal int SetupInterfaceList(mDNS *const m)
                     if (firstLoopback == NULL)
                         firstLoopback = i;
                 }
-                else
+                else if (i->ifi_flags & (IFF_MULTICAST | IFF_BROADCAST))  // http://b/25669326
                 {
                     if (SetupOneInterface(m, i->ifi_addr, i->ifi_netmask, i->ifi_name, i->ifi_index) == 0)
                         if (i->ifi_addr->sa_family == AF_INET)
-- 
2.19.2

